metis-remove-data/src/main/java/eu/europeana/metis/remove/discover/AbstractOrphanIdentification.java

```diff
@@ -37,19 +38,29 @@
   enum DiscoveryMode {
 
     /**
-     * This mode removes all orphans and their descendants.
+     * This mode discovers only identified orphans or their descendants that have no children. The
+     * deleted status of the orphan or its children is not taken into account. Note: in case that
+     * we're removing orphans, we should only use this mode if we intend to completely remove
+     * orphans from the history (rather than mark them as deleted).
      */
-    DISCOVER_ALL_ORPHANS(node -> true),
+    ORPHANS_WITHOUT_DESCENDANTS(node -> node.getChildren().isEmpty(), false),
 
     /**
-     * This mode removes only orphans or their descendants that have no children.
+     * This mode discovers only identified orphans or their descendants that have no children or
+     * only children that are marked as deleted (through {@link eu.europeana.metis.core.workflow.plugins.DataStatus#DELETED}).
+     * The node itself is not allowed to be deleted. Note: in case that we're removing orphans, we
+     * should only use this mode if we intend to mark orphans as deleted (rather than actually
+     * delete the history).
      */
-    DISCOVER_ONLY_CHILDLESS_ORPHANS(node -> node.getChildren().isEmpty());
+    ORPHANS_WITH_ONLY_DELETED_DESCENDANTS(node -> node.getChildren().isEmpty(), true);
 
-    private final Predicate<ExecutionPluginNode> test;
+    private final Predicate<ExecutionPluginNode> orphanTest;
```

Comment:
Does this have to be a parameter since it's the same for all items in the enum?

Reply:
True, it was needed for an unused DiscoveryMode that I have subsequently removed. I thought to leave it in place for now, but I can also remove it. What do you think?